tests/fs/test_localfs.py

```diff
@@ -119,3 +119,12 @@ def test_walk_detail(dir_path):
                 exp_root, basename
             )
             assert files[basename]["type"] == "file"
+
+
+@pytest.mark.skipif(
+    os.name == "nt", reason="A file name can't contain newlines on Windows"
```

Comment:
If windows doesn't support it - we should probably not support it either, as we are a cross-platform tool. Even though \n is acceptable in posix, it seems like we should not support it. At the same time the task of gathering a spec of supported/unsupported chars is a challenging one 🙁

Though alternative view might be to indeed support odd stuff like this, but convert it on windows somehow. I imagine that even on linux there are filesystems that might not be happy seeing newlines in the filenames. Case-sensitive/insensitive filesystems come to my mind when I think about this and how git tries to addapt the filenames during checkout.

That being said, using normpath directly is a reasonable thing to do for localfs, so I'm not against the change itself.

Reply:
How does git behave?

Reply:
git lets you commit whatever filename you want as long as your current system allows it. git also assumes that anyone checking out your repo will also be on a system that allows filenames which have already been committed. So you can use newlines in filenames on linux, but it will break when you try to checkout on windows.

If we are going to enforce that we only support paths that are valid on windows to maintain cross platform compatibility for DVC repos, `\/:*?"<>` are all also reserved/invalid characters for windows paths.

Reply:
IMO we should behave the same as git, and allow users to use whatever filenames they want, as long as we can actually quote/escape them properly in yaml dvc files. If a user needs their repo to be usable cross-platform, they should already be using windows-safe filenames.

Reply:
@pmrowla Great points. Makes total sense. Let's run with this PR then as is. Thank you.